PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java

```diff
@@ -21,188 +21,197 @@
+		gridData = new GridData();
+		gridData.horizontalAlignment = SWT.FILL;
+		gridData.grabExcessHorizontalSpace = true;
+		passwordField.setLayoutData(gridData);
+
+		return super.createDialogArea(parent);
+	}
+
+	@Override
+	protected void okPressed() {
+		synchronized (AddNewClusterForm.class) {
+			isCarryOnNextStep = true;
+			setErrorMessage(null);
+
+			AppInsightsClient.create(Messages.HDInsightAddNewClusterAction, null);
+
+			String clusterNameOrUrl = clusterNameField.getText().trim();
+			userName = userNameField.getText().trim();
+			storageName = storageNameField.getText().trim();
+
+			storageKey = storageKeyField.getText().trim();
+
+			password = passwordField.getText();
+
+			if (StringHelper.isNullOrWhiteSpace(clusterNameOrUrl) || StringHelper.isNullOrWhiteSpace(storageName)
+					|| StringHelper.isNullOrWhiteSpace(storageKey) || StringHelper.isNullOrWhiteSpace(userName)
+					|| StringHelper.isNullOrWhiteSpace(password)) {
+				setErrorMessage("Cluster Name, Storage Key, User Name, or Password shouldn't be empty");
+				isCarryOnNextStep = false;
+			} else {
+				clusterName = getClusterName(clusterNameOrUrl);
+
+				if (clusterName == null) {
+					setErrorMessage("Wrong cluster name or endpoint");
+					isCarryOnNextStep = false;
+				} else {
+					int status = ClusterManagerEx.getInstance().isHDInsightAdditionalStorageExist(clusterName,
+							storageName);
+					if (status == 1) {
+						setErrorMessage("Cluster already exist in current list");
+						isCarryOnNextStep = false;
+					} else if (status == 2) {
+						setErrorMessage("Default storage account is required");
+						isCarryOnNextStep = false;
+					}
+				}
+			}
+
+			if (isCarryOnNextStep) {
+				getStorageAccount();
+			}
+
+			if (isCarryOnNextStep) {
+				if (storageAccount != null) {
+					HDInsightAdditionalClusterDetail hdInsightAdditionalClusterDetail = new HDInsightAdditionalClusterDetail(
+							clusterName, userName, password, storageAccount);
+					ClusterManagerEx.getInstance().addHDInsightAdditionalCluster(hdInsightAdditionalClusterDetail);
+					hdInsightModule.refreshWithoutAsync();
+				}
+				super.okPressed();
+			}
+		}
+	}
+
+	// format input string
+	private static String getClusterName(String userNameOrUrl) {
+		if (userNameOrUrl.startsWith(URL_PREFIX)) {
+			return StringHelper.getClusterNameFromEndPoint(userNameOrUrl);
+		} else {
+			return userNameOrUrl;
+		}
+	}
+
+	private void getStorageAccount() {
+		PluginUtil.showBusy(true, getShell());
+		Display.getDefault().syncExec(new Runnable() {
+			@Override
+			public void run() {
+				try {
+					storageAccount = AddHDInsightAdditionalClusterImpl.getStorageAccount(clusterName, storageName,
+							storageKey, userName, password);
+					isCarryOnNextStep = true;
+				} catch (AzureCmdException | HDIException e) {
+					isCarryOnNextStep = false;
+					setErrorMessage(e.getMessage());
+				}
+			}
+		});
+		PluginUtil.showBusy(false, getShell());
+	}
```

Comment:
It might be better to remove white spaces change for the file to see what's the real changes.